### PR TITLE
Feature/notification api

### PR DESCRIPTION
**Pull request by MrZyr0**

### CODE REVIEW

**Review by Laroche-Nicolas — APPROVED** [review]

File: src/notification/notification.controller.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { Body, Controller, Get, Post } from '@nestjs/common';
+import { ApiTags } from '@nestjs/swagger';
+import { NotificationService } from './notification.service';
+import { SaveSubscriptionDto } from './save-subscription.dto';
+
+@ApiTags('Application Action')
```
Comment (MaxenceMottard) [inline src/notification/notification.controller.ts]: rename this tag

File: src/app.module.ts [hunk 2]
```diff
@@ -4,13 +4,16 @@ import { MongooseModule } from './mongoose.module';
...
 import config from './config';
+import { NotificationModule } from './notification/notification.module';
 
 @Module({
   imports: [
     PartyModule,
     MongooseModule,
     NestJSMongooseModule.forRoot(config().mongoUrl),
     UserModule,
+    ActionModule,
```
Comment (MaxenceMottard) [inline src/app.module.ts]: remove this line I've removed it

File: src/main.ts [hunk 3]
```diff
@@ -18,6 +18,9 @@ const useSwagger = (app: INestApplication) => {
 
 const bootstrap = async () => {
   const app = await NestFactory.create(AppModule);
+  app.enableCors({
```
Comment (MaxenceMottard) [inline src/main.ts]: please add condition to enable only if : !process.env.NODE_ENV || process.env.NODE_ENV !== 'production'

File: src/notification/notification.controller.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+@Controller('application')
```
Comment (MaxenceMottard) [inline src/notification/notification.controller.ts]: 🤮

File: src/notification/notification.controller.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    await this.applicationService.saveSubscription(body);
+    await this.applicationService.sendNotification({
+      userId: body.userId,
+      title: 'Backend notification registration successful',
+      message:
+        "Ceci est un test de notification retournée pour signaler que l'utilisateur est bien enregistré au stream des notifs",
+    });
+  }
+
+  @Post('test-notif')
```
Comment (MaxenceMottard) [inline src/notification/notification.controller.ts:27]: maybe remove this useless route ?

File: src/notification/notification.service.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import { NotificationSubscriptionRepository } from '../repositories/notification-subscription.repository';
+import { SaveSubscriptionDto } from './save-subscription.dto';
+
+@Injectable()
+export class NotificationService {
+  constructor(
+    private readonly notificationSubscriptionRepository: NotificationSubscriptionRepository,
+  ) {
+    webpush.setVapidDetails(
+      'mailto:test@test.com',
```
Comment (MaxenceMottard) [inline src/notification/notification.service.ts:13]: replace mail with: julien.seixas@ynov.com just to piss him off

File: src/notification/notification.service.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  sendNotification = async ({
+    userId,
+    title,
+    message,
+  }: Record<string, string>) => {
+    try {
+      const notificationSubscriptions =
+        await this.notificationSubscriptionRepository.findManyBy({ userId });
+      const notificationData = { title, message };
+      // const notifications: Promise<SendResult>[] = [];
```
Comment (MaxenceMottard) [inline src/notification/notification.service.ts]: remove

**Review by MaxenceMottard — CHANGES_REQUESTED** [review]

File: src/notification/notification.controller.ts (hunk 5 above)
Comment (Laroche-Nicolas) [reply src/notification/notification.controller.ts:27]: No we keep it for a POC

**Review by MaxenceMottard — APPROVED** [review]

**Pull request merged**